fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/libsumo/SimulationSimulateStep.java

```diff
@@ -66,31 +86,41 @@ public SimulationSimulateStep() {
         readTrafficLights(results);
 
         return results;
+
     }
 
     private void readVehicles(List<AbstractSubscriptionResult> results) {
-        String mosaicVehicleId;
-        for (String sumoVehicleId : Vehicle.getIDList()) {
-            mosaicVehicleId = Bridge.VEHICLE_ID_TRANSFORMER.fromExternalId(sumoVehicleId);
+        for (String arrived : Simulation.getArrivedIDList()) {
+            VEHICLE_SUBSCRIPTIONS.remove(Bridge.VEHICLE_ID_TRANSFORMER.fromExternalId(arrived));
```

Comment:
Isn't this operation way faster when using a `HashSet`?

Reply:
yes, but hashset is quite bad for iterating trough, though :<

Reply:
Oh, and arrived vehicles are usually more at the beginning of the subscriptions-list 😄